File: Pages/DataTablePage.cs
using OpenQA.Selenium;
using OpenQA.Selenium.Support.UI;
using OpenQA.Selenium.Support.Extensions;

public class DataTablePage
{
    private readonly IWebDriver driver;
    private readonly WebDriverWait wait;

    public DataTablePage(IWebDriver driver)
    {
        this.driver = driver;
        wait = new WebDriverWait(driver, TimeSpan.FromSeconds(10));
    }

    private IWebElement RecruitmentPromotionElement => wait.Until(driver => driver.FindElement(By.XPath("//a[contains(@class,'datalist_link font-onesix-bold stat-content-space')][normalize-space()='Recruitment and promotion']")));
    private IWebElement ManagersWithPlElement => wait.Until(driver => driver.FindElement(By.XPath("//a[contains(text(),'Managers with')]")));
    private IWebElement PmAndIcElement => wait.Until(driver => driver.FindElement(By.XPath("//a[contains(text(),'People managers and individual contributors')]")));
    private IWebElement EmploymentElement => wait.Until(driver => driver.FindElement(By.XPath("//a[contains(text(),'Employment')]")));
    private IWebElement BoardCompositionElement => wait.Until(driver => driver.FindElement(By.XPath("(//a[contains(text(),'Board composition')])[2]")));
    private IWebElement EqualPayElement => wait.Until(driver => driver.FindElement(By.XPath("//a[normalize-space()='Equal pay']")));
    private IWebElement HighPotentialsElement => wait.Until(driver => driver.FindElement(By.XPath("//a[contains(text(),'High potentials')]")));
    private IWebElement PerformanceElement => wait.Until(driver => driver.FindElement(By.XPath("//a[normalize-space()='Performance']")));
    private IWebElement LeadershipTrainingElement => wait.Until(driver => driver.FindElement(By.XPath("//a[normalize-space()='Leadership development training']")));
    private IWebElement MentoringElement => wait.Until(driver => driver.FindElement(By.XPath("//a[contains(text(),'Mentoring')]")));
    private IWebElement ParentalLeaveElement => wait.Until(driver => driver.FindElement(By.XPath("//a[contains(text(),'Parental leave')]")));


    private IWebElement SubmitButton => wait.Until(driver => driver.FindElement(By.XPath("//a[contains(text(),'Submit')]")));

    public void ClickRecruitmentPromotion()
    {
        RecruitmentPromotionElement.Click();
    }
    public void ClickManagersWithPL()
    {
        ManagersWithPlElement.Click(); 
    }
    public void ClickPmAndIc()
    {
        PmAndIcElement.Click();
    }
    public void ClickEmployment()
    { 
        EmploymentElement.Click();
    }
    public void ClickBoardComposition()
    {
        BoardCompositionElement.Click();
    }
    public void ClickEqualPay()
    {
        EqualPayElement.Click();
    }
    public void ClickHighPotentials()
    {
        HighPotentialsElement.Click();
    }
    public void ClickPerformance()
    {
        PerformanceElement.Click();
    }
    public void ClickLeadershipDevelopmentTraining()
    {
        LeadershipTrainingElement.Click();
    }
    public void ClickMentoring()
    {
        MentoringElement.Click();
    }
    public void ClickParentalLeave()
    {
        ParentalLeaveElement.Click();   
    }
    public void HandleAlert()
    {
        // Wait for the alert to be present using Lambda Expression
        wait.Until(driver => driver.SwitchTo().Alert());

        // Switch to the alert
        IAlert alert = driver.SwitchTo().Alert();

        // Click on OK button
        alert.Accept();
    }
    public void ClickSubmitButton()
    {
        SubmitButton.Click();
    }
}

File: pages/data-table.page.ts
import { type Page, type Locator } from '@playwright/test';

/**
 * Page Object Model for the Data Table / Data Lists page.
 * Handles navigation through all data list sections and form submission.
 */
export class DataTablePage {
  // Data list section locators
  readonly recruitmentPromotion: Locator;
  readonly managersWithPl: Locator;
  readonly pmAndIc: Locator;
  readonly employment: Locator;
  readonly boardComposition: Locator;
  readonly equalPay: Locator;
  readonly highPotentials: Locator;
  readonly performance: Locator;
  readonly leadershipTraining: Locator;
  readonly mentoring: Locator;
  readonly sponsorship: Locator;
  readonly parentalLeave: Locator;

  // Action locators
  readonly submitButton: Locator;

  constructor(private page: Page) {
    this.recruitmentPromotion = page.locator(
      "a.datalist_link.font-onesix-bold.stat-content-space:has-text('Recruitment and promotion')"
    );
    this.managersWithPl = page.locator("a:has-text('Managers with')");
    this.pmAndIc = page.locator(
      "a:has-text('People managers and individual contributors')"
    );
    this.employment = page.locator("a:has-text('Employment')");
    this.boardComposition = page.locator(
      "a:has-text('Board composition') >> nth=1"
    );
    this.equalPay = page.locator("a.datalist_link:has-text('Equal pay')");
    this.highPotentials = page.locator("a:has-text('High potentials')");
    this.performance = page.locator("a:has-text('Performance')");
    this.leadershipTraining = page.locator(
      "a:has-text('Leadership development training')"
    );
    this.mentoring = page.locator("a:has-text('Mentoring')");
    this.sponsorship = page.locator("a:has-text('Sponsorship')");
    this.parentalLeave = page.locator("a:has-text('Parental leave')");

    this.submitButton = page.locator("a:has-text('Submit')");
  }

  // --- Click methods for each data list section ---

  async clickRecruitmentPromotion(): Promise<void> {
    await this.recruitmentPromotion.click();
  }

  async clickManagersWithPL(): Promise<void> {
    await this.managersWithPl.click();
  }

  async clickPmAndIc(): Promise<void> {
    await this.pmAndIc.click();
  }

  async clickEmployment(): Promise<void> {
    await this.employment.click();
  }

  async clickBoardComposition(): Promise<void> {
    await this.boardComposition.click();
  }

  async clickEqualPay(): Promise<void> {
    await this.equalPay.click();
  }

  async clickHighPotentials(): Promise<void> {
    await this.highPotentials.click();
  }

  async clickPerformance(): Promise<void> {
    await this.performance.click();
  }

  async clickLeadershipDevelopmentTraining(): Promise<void> {
    await this.leadershipTraining.click();
  }

  async clickMentoring(): Promise<void> {
    await this.mentoring.click();
  }

  async clickSponsorship(): Promise<void> {
    await this.sponsorship.click();
  }

  async clickParentalLeave(): Promise<void> {
    await this.parentalLeave.click();
  }

  // --- Action methods ---

  /**
   * Click the Submit button.
   */
  async clickSubmitButton(): Promise<void> {
    await this.submitButton.click();
  }

  /**
   * Register a one-time dialog handler to accept the next alert/confirm dialog.
   * IMPORTANT: Call this BEFORE the action that triggers the dialog.
   */
  async handleAlert(): Promise<void> {
    this.page.once('dialog', async (dialog) => {
      await dialog.accept();
    });
  }
}
